src/azure-cli-core/azure/cli/core/azclierror.py

```diff
@@ -63,6 +64,7 @@ def print_error(self):
 
     def send_telemetry(self):
         telemetry.set_error_type(self.__class__.__name__)
+        log_latest_error_info(self.error_msg, self.__class__.__name__)
```

Comment:
Will `log_latest_error_info` be called if telemetry is turned off by setting `core.collect_telemetry=false`?

Reply:
The reason why the exception log is recorded here is to record the same exception information locally as that sent to the telemetry for subsequent matching.
I will follow up with @houk-ms  to see if there is a more suitable place to record all exceptions uniformly.

Reply:
This is a valid concern, when setting `core.collect_telemetry=false`. The function `send_telemetry` will be called,  but the telemtry record won't be written to cache file. This is controlled in https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/telemetry.py#L235. While in this case `log_latest_error_info` will log the error info. 

Another problem is this is not the only entry point where we log telemetry.

So as we synced offline, we may utilize telemetry/cache file to avoid the problem.

Reply:
Yes, this is a great suggestion. It has been added to issue #16770